src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/SwaggerGenServiceCollectionExtensions.cs

```diff
@@ -70,21 +94,40 @@ private JsonSerializerOptions ResolveOptions()
             JsonSerializerOptions serializerOptions;
 
             /*
-             * First try to get the options configured for MVC,
-             * then try to get the options configured for Minimal APIs if available,
-             * then try the default JsonSerializerOptions if available,
-             * otherwise create a new instance as a last resort as this is an expensive operation.
+             * There is no surefire way to do this.
+             * However, both JsonOptions are defaulted in the same way.
+             * If neither is configured it makes no difference which one is chosen.
+             * If both are configured, then we just need to make a choice.
+             * As Minimal APIs are newer if someone is configuring them
+             * it's probably more likely that is what they're using.
+             * 
+             * If either JsonOptions is null we will try to create a new instance as
+             * a last resort as this is an expensive operation.
              */
 #if NET
             serializerOptions =
-                _serviceProvider.GetService<IOptions<AspNetCore.Mvc.JsonOptions>>()?.Value?.JsonSerializerOptions
-                ?? _serviceProvider.GetService<IOptions<AspNetCore.Http.Json.JsonOptions>>()?.Value?.SerializerOptions
+                _serviceProvider.GetService<IOptions<AspNetCore.Http.Json.JsonOptions>>()?.Value?.SerializerOptions
                 ?? JsonSerializerOptions.Default;
+
+            if (HasConfiguredMinimalApiJsonOptions())
+            {
+                serializerOptions ??= _serviceProvider.GetService<IOptions<AspNetCore.Http.Json.JsonOptions>>()?.Value?.SerializerOptions;
+            }
 #else
             serializerOptions = new JsonSerializerOptions();
 #endif
```

Comment:
I wonder if a better thing to do would be:

1. Add a new first-class object (just-like the `JsonOptions`) to hold Swashbuckle's own `JsonSerializerOptions`. Users can then specifically register that if they wish to.
2. We add a `IPostConfigureOptions<T>` for that type that applies the above heuristics if the options in that class are null at the point of being resolved.
3. Then, the new overloads just provide a bridge from one `JsonOptions` to that new type. Then we don't need to replace services.

Reply:
Instead of or in addition to these methods?

Reply:
In addition as syntactic-sugar.